### PR TITLE
[dogstatsd] capture: make capture destination location configurable

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR simply aims to make the destination location for dogstatsd captures configurable. Should no location be specified, we will default to the current behavior (typically `/opt/datadog-agent/run/dsd_capture`). However, if a destination path is specified, the capture should be written to it.

### Motivation

This should make the feature more usable and useful on production environment where we might want to write the captured file to a mounted volume, etc.

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Simply run a capture as follows:
```
./bin/agent/agent -c ./bin/agent/dist/datadog.yaml dogstatsd-capture -d 60s -p /some/path/to/test
```

The file should be written to the expected location as opposed to the default.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Comment by vickenty** [comment]
If I read the code right, this allows users with access to agent token to create arbitrary paths and overwrite arbitrary files as the user that the agent runs under (possibly root). At least, we should refuse to create anything anywhere that isn't world-writeable already. Better would be to stream the data to the caller process, so they can write it to the disk using their own privileges.

**Comment by truthbk** [comment]
@vickenty the agent does not run as root unless the user explicitly changes the default. The agent process, which writes the capture file, typically writes as the `dd-agent` user, any location that is not `dd-agent` writable will fail to create the file, an error will be returned when the file descriptor is attempted to be opened, and the API will return that error. 

At high-throughputs streaming the content back over the API is non-trivial and can have serious IO impact. I see the value, but I believe that would be another feature altogether. 

Feel free to reach out if you wish to discuss further.

File: cmd/agent/app/dogstatsd_capture.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ const (
 func init() {
 	AgentCmd.AddCommand(dogstatsdCaptureCmd)
 	dogstatsdCaptureCmd.Flags().DurationVarP(&dsdCaptureDuration, "duration", "d", defaultCaptureDuration, "Duration traffic capture should span.")
+	dogstatsdCaptureCmd.Flags().StringVarP(&dsdCaptureFilePath, "path", "p", "", "Directory path to write the captue to.")
```
Comment (ogaca-dd) [inline cmd/agent/app/dogstatsd_capture.go:42]:
```suggestion
	dogstatsdCaptureCmd.Flags().StringVarP(&dsdCaptureFilePath, "path", "p", "", "Directory path to write the capture to.")
```

File: pkg/dogstatsd/replay/writer.go [hunk 2]
```diff
@@ -70,10 +71,9 @@ type TrafficCaptureWriter struct {
 }
 
 // NewTrafficCaptureWriter creates a TrafficCaptureWriter instance.
-func NewTrafficCaptureWriter(l string, depth int) *TrafficCaptureWriter {
+func NewTrafficCaptureWriter(depth int) *TrafficCaptureWriter {
 
 	return &TrafficCaptureWriter{
-		Location:    l,
 		Traffic:     make(chan *CaptureBuffer, depth),
```
Comment (ogaca-dd) [inline pkg/dogstatsd/replay/writer.go:77]: `Location` field can be removed from `TrafficCaptureWriter`.

**Review by ogaca-dd — APPROVED** [review]
LGTM

File: pkg/dogstatsd/replay/writer.go (hunk 2 above)
Comment (truthbk) [reply pkg/dogstatsd/replay/writer.go:77]: It's actually still used, but it's set when `Capture()` is called. I'll keep it. Thanks for keeping me honest. ;)

Comment (ogaca-dd) [reply pkg/dogstatsd/replay/writer.go:77]: Isn't `Location ` be used only in the function `Capture()`?

**Comment by vickenty** [comment]
> @vickenty the agent does not run as root unless the user explicitly changes the default. The agent process, which writes the capture file, typically writes as the `dd-agent` user, any location that is not `dd-agent` writable will fail to create the file, an error will be returned when the file descriptor is attempted to be opened, and the API will return that error.

The fact that we don't run the agent as root by default doesn't mean that no one does. Even if not running as root, this still allows lateral shift in privileges, depending on customer setup, and this is treated as a security vulnerability in other software (google for "arbitrary file write cve", for example).
 
> At high-throughputs streaming the content back over the API is non-trivial and can have serious IO impact. I see the value, but I believe that would be another feature altogether.

`lo` has higher throughput than almost any drive commonly available, if blocks are large enough the performance penalty should be manageable. If not running over loopback, this is remote file write. But I agree that adds complexity and might be non-trivial to implement, so I think we should either stick to either fixed or world-writeable location.

File: pkg/dogstatsd/replay/writer.go (hunk 2 above)
Comment (truthbk) [reply pkg/dogstatsd/replay/writer.go:80]: Yes, but I think I want to keep it around regardless. If you have a strong opinion I can remove it.

**Review by vickenty — APPROVED** [review]

**Pull request merged**